store/postgres/src/relational_queries.rs

```diff
@@ -2617,6 +2659,108 @@ impl<'a> SortKey<'a> {
             }
         }
 
+        fn with_child_object_key<'a>(
+            parent_table: &'a Table,
+            child_table: &'a Table,
+            join_attribute: String,
+            derived: bool,
+            attribute: String,
+            direction: &'static str,
+        ) -> Result<SortKey<'a>, QueryExecutionError> {
+            let sort_by_column = child_table.column_for_field(&attribute)?;
+            if sort_by_column.is_fulltext() {
+                Err(QueryExecutionError::NotSupported(
+                    "Sorting by fulltext fields".to_string(),
+                ))
+            } else if sort_by_column.is_primary_key() {
+                Err(QueryExecutionError::NotSupported(
+                    "Sorting by id".to_string(),
+                ))
+            } else {
+                let (parent_column, child_column) = match derived {
+                    true => (
+                        parent_table.primary_key(),
+                        child_table
+                            .column_for_field(&join_attribute)
+                            .expect("Column for a join attribute not found"),
```

Comment:
The `expect` here (and all `expects`) should be turned into a `constraint_violation!`, otherwise queries that somehow make it here causes panics which are much more expensive than returning errors.

Reply:
34d214e05